plugins/database/src/db_plugin.cpp

```diff
@@ -15792,6 +15794,190 @@ auto db_get_delay_rule_info_op(
+
+            sql += " where data_id = ? and resc_id = ?";
+
+            irods::log(LOG_DEBUG8, fmt::format("statement:[{}]", sql));
+            irods::log(LOG_DEBUG9, fmt::format("before:[{}]", before.dump()));
+            irods::log(LOG_DEBUG9, fmt::format("after:[{}]", after.dump()));
+
+            // We have to capture the strings first so that binding to the cllBindVars does not
+            // rely on temporary variables.
+            data_repl_num = after.at("data_repl_num").get<std::string>();
+            data_version = after.at("data_version").get<std::string>();
+            data_type_name = after.at("data_type_name").get<std::string>();
+            data_size = after.at("data_size").get<std::string>();
+            data_path = after.at("data_path").get<std::string>();
+            data_owner_name = after.at("data_owner_name").get<std::string>();
+            data_owner_zone = after.at("data_owner_zone").get<std::string>();
+            data_is_dirty = after.at("data_is_dirty").get<std::string>();
+            data_status = after.at("data_status").get<std::string>();
+            data_checksum = after.at("data_checksum").get<std::string>();
+            data_expiry_ts = after.at("data_expiry_ts").get<std::string>();
+            data_map_id = after.at("data_map_id").get<std::string>();
+            data_mode = after.at("data_mode").get<std::string>();
+            r_comment = after.at("r_comment").get<std::string>();
+            create_ts = after.at("create_ts").get<std::string>();
+            modify_ts = after.at("modify_ts").get<std::string>();
+            resc_id = after.at("resc_id").get<std::string>();
+
+            where_data_id = before.at("data_id").get<std::string>();
+            where_resc_id = before.at("resc_id").get<std::string>();
+
+            // Assign bind variables for execution shortly...
+            cllBindVars[cllBindVarCount++] = data_repl_num.data();
+            cllBindVars[cllBindVarCount++] = data_version.data();
+            cllBindVars[cllBindVarCount++] = data_type_name.data();
+            cllBindVars[cllBindVarCount++] = data_size.data();
+            cllBindVars[cllBindVarCount++] = data_path.data();
+            cllBindVars[cllBindVarCount++] = data_owner_name.data();
+            cllBindVars[cllBindVarCount++] = data_owner_zone.data();
+            cllBindVars[cllBindVarCount++] = data_is_dirty.data();
+            cllBindVars[cllBindVarCount++] = data_status.data();
+            cllBindVars[cllBindVarCount++] = data_checksum.data();
+            cllBindVars[cllBindVarCount++] = data_expiry_ts.data();
+            cllBindVars[cllBindVarCount++] = data_map_id.data();
+            cllBindVars[cllBindVarCount++] = data_mode.data();
+            cllBindVars[cllBindVarCount++] = r_comment.data();
+            cllBindVars[cllBindVarCount++] = create_ts.data();
+            cllBindVars[cllBindVarCount++] = modify_ts.data();
+            cllBindVars[cllBindVarCount++] = resc_id.data();
```

Comment:
I would guess `.data()` probably because this was originally written before our `.c_str()` only directive. `cllBindVars` is an array of `const char*`.

Reply:
Ok. Let's use c_str() instead. Just adds more protection and better expresses intent.

Reply:
Done